pkg/plugins/common/kustomize/v2-alpha/scaffolds/init.go

```diff
@@ -62,23 +62,45 @@ func (s *initScaffolder) Scaffold() error {
 		machinery.WithConfig(s.config),
 	)
 
-	return scaffold.Execute(
-		&rbac.Kustomization{},
-		&rbac.AuthProxyRole{},
-		&rbac.AuthProxyRoleBinding{},
-		&rbac.AuthProxyService{},
-		&rbac.AuthProxyClientRole{},
-		&rbac.RoleBinding{},
-		&rbac.LeaderElectionRole{},
-		&rbac.LeaderElectionRoleBinding{},
-		&rbac.ServiceAccount{},
-		&manager.Kustomization{},
-		&manager.Config{Image: imageName},
-		&manager.ControllerManagerConfig{},
-		&kdefault.Kustomization{},
-		&kdefault.ManagerAuthProxyPatch{},
-		&kdefault.ManagerConfigPatch{},
-		&prometheus.Kustomization{},
-		&prometheus.Monitor{},
-	)
+	if s.config.IsComponentConfig() {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&manager.ControllerManagerConfig{},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
+			&prometheus.Monitor{},
+		)
+	} else {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
+			&prometheus.Monitor{},
+		)
+
+	}
```

Comment:
I would do the same thing here in v2-alpha as the comment in the v1 init.go change.

Reply:
Link to comment in v1: https://github.com/kubernetes-sigs/kubebuilder/pull/2826#discussion_r939060615

Reply:
Agree and Modified accordingly.